subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ResolvedFilesCollectingVisitor.java

```diff
@@ -33,8 +33,12 @@
 
     @Override
     public void visitArtifact(AttributeContainer variant, ResolvedArtifact artifact) {
-        // Defer adding the artifacts until after all the file dependencies have been visited
-        this.artifacts.add(artifact);
+        try {
+            artifact.getFile(); // triggering file resolve
+            this.artifacts.add(artifact);
+        } catch (Throwable t) {
+            failures.add(t);
```

Comment:
Good catch. The code further down was actually never triggered. Since we do not add the artifact to `this.artifacts` here if a an exception occurs, it won't be considered further down. I.e., no resolving is triggered there.

I removed the `try / catch` further down.